#!/usr/bin/env python
# field of view configuration

PACKAGE="pcd_tutorial"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("leafsize", double_t, 1, "Leafsize", 0.02, 0.002, 0.3)
gen.add("MeanK", int_t, 1, "MeanK", 50, 1, 500)
gen.add("StddevMulThresh", double_t, 1, "StddevMulThresh", 0.5, 0.0, 10.0)
gen.add("overlap_threshold", double_t, 1, "Overlap_threshold", 0.6, 0.001, 1)
gen.add("big_plane_size", int_t, 1, "big_plane_size", 600, 10, 10000)
gen.add("min_cloud_size", int_t, 1, "min_cloud_size", 50, 10, 10000)
gen.add("hv_tolerance", double_t, 1, "hv_tolerance", 15.0, 0.0, 90.0)
gen.add("lean_tolerance", double_t, 1, "lean_tolerance", 5.0, 0.0, 90.0)
gen.add("DistanceThresholdPlane0", double_t, 1, "DistanceThresholdPlane0", 0.01, 0.0, 5.0)
gen.add("NormalDistanceWeightPlane", double_t, 1, "NormalDistanceWeightPlane", 0.035, 0.0, 5.0)
gen.add("MaxIterationsPlane", int_t, 1, "MaxIterationsPlane", 50, 1, 100000)
gen.add("DistanceThresholdPlane", double_t, 1, "DistanceThresholdPlane", 0.01, 0.0, 5.0)


exit(gen.generate(PACKAGE, "pcd_tutorial", "pcd_tutorial_node"))
